messaging/src/main/java/org/axonframework/eventhandling/DefaultEventProcessorSpanFactory.java

```diff
@@ -0,0 +1,199 @@
+     * Creates a root trace for each batch of events and a linked handler span for each event in the batch by default. This
+     * behavior can be changed by setting {@link Builder#disableBatchTrace(boolean)} to {@code true}.
+     * <p>
+     * If you prefer to have asynchronously handled events in the same trace as the trace that published it, you can set
+     * {@link Builder#distributedInSameTrace(boolean)} to {@code true}. If {@code true}, this duration defaults to 2
+     * minutes, which you can adjust using {@link Builder#distributedInSameTraceTimeLimit(Duration)}.
+     *
+     * @author Mitchell Herrijgers
+     * @since 4.9.0
+     */
+    public class DefaultEventProcessorSpanFactory implements EventProcessorSpanFactory {
+
+        private final SpanFactory spanFactory;
+        private final boolean disableBatchTrace;
+        private final boolean distributedInSameTrace;
+        private final Duration distributedInSameTraceTimeLimit;
+
+        /**
+         * Creates a new {@link DefaultEventProcessorSpanFactory} using the provided {@code builder}.
+         *
+         * @param builder The builder to build the {@link DefaultEventProcessorSpanFactory} from.
+         */
+        protected DefaultEventProcessorSpanFactory(Builder builder) {
+            builder.validate();
+            this.spanFactory = builder.builderSpanFactory;
+            this.disableBatchTrace = builder.builderDisableBatchTrace;
+            this.distributedInSameTrace = builder.builderDistributedInSameTrace;
+            this.distributedInSameTraceTimeLimit = builder.builderDistributedInSameTraceTimeLimit;
+        }
+
+
+        /**
+         * Creates a new {@link Builder} to build a {@link DefaultEventProcessorSpanFactory} with. The default values are:
+         * <ul>
+         *     <li>{@code disableBatchTrace} defaults to {@code false}</li>
+         *     <li>{@code distributedInSameTrace} defaults to {@code false}</li>
+         *     <li>{@code distributedInSameTraceTimeLimit} defaults to 2 minutes</li>
+         * </ul>
+         * The {@code spanFactory} is a required field and should be provided.
+         *
+         * @return The {@link Builder} to build a {@link DefaultEventProcessorSpanFactory} with.
+         */
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        @Override
+        public Span createBatchSpan(boolean streaming, List<? extends EventMessage<?>> eventMessages) {
+            if (distributedInSameTrace || disableBatchTrace || !streaming) {
+                return NoOpSpanFactory.NoOpSpan.INSTANCE;
+            }
+            return spanFactory.createRootTrace(() -> "StreamingEventProcessor.batch");
+        }
+
+        @Override
+        public Span createHandleEventSpan(boolean streaming, EventMessage<?> eventMessage) {
+            if (!streaming) {
+                return spanFactory.createChildHandlerSpan(() -> "SubscribingEventProcessor.handle", eventMessage);
```

Comment:
If I follow correctly, this changes the Subscribing processor's span from an internal-span to a childer handler span.
Although I think the change makes sense, just validating whether that was intended.

Reply:
Don't we want just the interface's name in the span, just like most other span names? So just "EventProcessor.handle"? It seems inconsistent now.

Reply:
Yes that change was intended. It's part of a larger transaction and is not part of a distributed/async action. So it should be part of the same trace. 

@gklijs Yeah it makes sense. I would be in favor of keeping `StreamingEventProcessor` and `EventProcessor` then, to keep them distinguished.